src/python/pants/build_graph/address_test.py

```diff
@@ -349,6 +345,34 @@ def assert_spec(address: Address, *, expected: str, expected_path_spec: str) ->
         expected_path_spec="a.b.subdir.dir2.f.txt@@b",
     )
 
+    assert_spec(
+        Address("", target_name="template", parameters={"k1": "v", "k2": "v"}),
+        expected="//:template@k1=v,k2=v",
+        expected_path_spec=".template@@k1=v,k2=v",
+    )
+    assert_spec(
+        Address("a/b", parameters={"k1": "v", "k2": "v"}),
+        expected="a/b@k1=v,k2=v",
+        expected_path_spec="a.b@@k1=v,k2=v",
+    )
+    assert_spec(
+        Address("a/b", generated_name="gen", parameters={"k": "v"}),
+        expected="a/b@k=v#gen",
```

Comment:
Note that I think this should be `a/b#gen@k=v`, which I think Stu recently agreed to. But saving that for later.